rust_snuba/src/processors/eap_spans.rs

```diff
@@ -119,32 +115,71 @@ impl From<FromSpanMessage> for EAPSpan {
                 &mut res.attr_str_~N,
                 )*
             ];
-            });
-
-            sentry_tags.iter().chain(tags.iter()).for_each(|(k, v)| {
-                attr_str_buckets[(fnv_1a(k.as_bytes()) as usize) % attr_str_buckets.len()]
-                    .insert(k.clone(), v.clone());
-            });
-        }
-
-        {
-            seq!(N in 0..20 {
             let mut attr_num_buckets = [
                 #(
                 &mut res.attr_num_~N,
                 )*
             ];
+            });
+
+            let mut insert_string = |k: String, v: String| {
+                attr_str_buckets[(fnv_1a(k.as_bytes()) as usize) % attr_str_buckets.len()]
+                    .insert(k.clone(), v.clone());
+            };
+
+            let mut insert_num = |k: String, v: f64| {
+                attr_num_buckets[(fnv_1a(k.as_bytes()) as usize) % attr_num_buckets.len()]
+                    .insert(k.clone(), v);
+            };
+
+            if let Some(sentry_tags) = from.sentry_tags {
+                sentry_tags.iter().for_each(|(k, v)| {
+                    if k == "transaction" {
+                        res.segment_name = v.clone();
+                    } else {
+                        insert_string(k.clone(), v.clone());
+                    }
+                })
+            }
+
+            if let Some(tags) = from.tags {
+                tags.iter().for_each(|(k, v)| {
+                    insert_string(k.clone(), v.clone());
+                })
+            }
+
+            if let Some(measurements) = from.measurements {
+                measurements.iter().for_each(|(k, v)| {
+                    if k == "client_sample_rate" && v.value != 0.0 {
```

Comment:
I think I would find this measurement specifically outside the loop as well.

Reply:
ditto